libdnf/sack/advisory.cpp

```diff
@@ -151,9 +152,59 @@ std::vector<AdvisoryModule> Advisory::getModules() const
         moduleList.emplace_back(sack, advisory, name, stream, version, context, arch);
     }
     dataiterator_free(&di);
+
     return moduleList;
 }
 
+std::vector<AdvisoryCollection> Advisory::getCollections() const
```

Comment:
What about to also add method, get applicable packages. From code I sow that after `getCollections()` there is an iterator that ask if collection is applicable and then it ask for collection packages.
```
                for (auto & advisoryCollection: advisory.getCollections()) {
                    if (advisoryCollection.isApplicable()) {
                        auto new_pkgs = advisoryCollection.getPackages();
```
The new approach could skip creation of collection object at all and generates only advisory packages for applicable collections. What do you think?

Reply:
I have used the approach we discussed on our meeting. The AdvisoryCollection class was removed. It is simpler and I think better approach.

I have also separated the unittest into separate files for Advisory, Query and DnfPackage.